Sources/SwiftInspectorCommands/Tests/TypeLocationCommandSpec.swift

```diff
@@ -0,0 +1,189 @@
+      }
+
+      describe("run") {
+
+        context("with no arguments") {
+          it("fails") {
+            let result = try? TestTypeLocationTask.run(path: nil, name: nil)
+            expect(result?.didFail) == true
+          }
+        }
+
+        context("path is valid") {
+          beforeEach { pathURL = try? Temporary.makeFile(content: "struct Foo { }") }
+
+          context("with no --name argument") {
+            it("fails") {
+              let result = try? TestTypeLocationTask.run(path: pathURL.path, name: nil)
+              expect(result?.didFail) == true
+            }
+          }
+
+          context("with an empty --name argument") {
+            it("fails") {
+              let result = try? TestTypeLocationTask.run(path: pathURL.path, name: "")
+              expect(result?.didFail) == true
+            }
+          }
+        }
+
+        context("name is valid") {
+          context("with no --path argument") {
+            it("fails") {
+              let result = try? TestTypeLocationTask.run(path: nil, name: "Foo")
+              expect(result?.didFail) == true
+            }
+          }
+
+          context("with an empty --path argument") {
+            it("fails") {
+              let result = try? TestTypeLocationTask.run(path: "", name: "Foo")
+              expect(result?.didFail) == true
+            }
+          }
+        }
+
+        context("when path is a directory") {
+          beforeEach { pathURL = try? Temporary.makeFolder() }
+
+          it("fails") {
+            let result = try? TestTypeLocationTask.run(path: pathURL.path, name: "Foo")
+            expect(result?.didFail) == true
+          }
+        }
+
+        context("when path is a file") {
+          beforeEach { pathURL = try? Temporary.makeFile(content: "struct Foo { }") }
+
+          it("succeeds") {
+            let result = try? TestTypeLocationTask.run(path: pathURL.path, name: "Foo")
+            expect(result?.didSucceed) == true
+          }
+
+          context("type is found") {
+            it("outputs the correct line numbers") {
+              let contents =
+              """
+              import Foundation
+
+              struct Foo { }
+              """
+
+              pathURL = try? Temporary.makeFile(content: contents)
+
+              let result = try? TestTypeLocationTask.run(path: pathURL.path, name: "Foo")
+              expect(result?.outputMessage).to(contain("2 2"))
+            }
+          }
+
+          context("type is not found") {
+            it("outputs nothing") {
+              let contents =
+              """
+              import Foundation
+
+              struct Foo { }
+              """
+
+              pathURL = try? Temporary.makeFile(content: contents)
+
+              let result = try? TestTypeLocationTask.run(path: pathURL.path, name: "Bar")
+              expect(result?.outputMessage).to(equal(""))
+            }
+          }
+
+          context("multiple types found") {
```

Comment:
I knowingly decided not to try to see what happens if a type of the same name is nested within another type of the same name.